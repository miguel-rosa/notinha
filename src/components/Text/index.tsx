import React, { FC } from "react";
import { StyleProp, TextStyle, Text as RNText} from "react-native";

type TextProps = {
  style?: StyleProp<TextStyle>;
  weight?: "500" | "700",
  fontSize?: number
}

const Text:FC<TextProps> = ({style, weight="500", children, fontSize}) => {
  const fontWeight = {
    "700": "Poppins_700Bold", 
    "500": "Poppins_500Medium"
  }
  return (
    <RNText style={[
      {
        fontFamily: fontWeight[weight],
        fontSize: fontSize
      },
      style
    ]}>
      {children}
    </RNText>
  )
}


export default Text;