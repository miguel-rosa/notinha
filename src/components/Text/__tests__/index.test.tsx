import React from "react";
import { render, fireEvent } from '@testing-library/react-native';

import Text from "../index";

const innerText = "Text Component"

describe("given a text, the component should render it as children", () => {
  it("is rendering the children", () => {
    const tree = render(<Text>{innerText}</Text>).toJSON();
    expect(tree.children[0]).toBe(innerText)
  })    
})

describe('CSS API', () => {
  it("font weight 500", () => {
    const tree = render(<Text weight="500">{innerText}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  })  
  it("font weight 700", () => {
    const tree = render(<Text weight="700">{innerText}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  })
});