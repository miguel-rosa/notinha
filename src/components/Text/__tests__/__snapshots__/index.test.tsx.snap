// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSS API font weight 500 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Poppins_500Medium",
        "fontSize": undefined,
      },
      undefined,
    ]
  }
>
  Text Component
</Text>
`;

exports[`CSS API font weight 700 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Poppins_700Bold",
        "fontSize": undefined,
      },
      undefined,
    ]
  }
>
  Text Component
</Text>
`;
